deps/spidershim/src/v8isolate.cc

```diff
@@ -313,4 +315,28 @@ void* Isolate::GetData(uint32_t slot) {
 TryCatch* Isolate::GetTopmostTryCatch() const { return pimpl_->topTryCatch; }
 
 void Isolate::SetTopmostTryCatch(TryCatch* val) { pimpl_->topTryCatch = val; }
+
+class SimpleJSRuntimeStats : public JS::RuntimeStats {
+  public:
+    explicit SimpleJSRuntimeStats(mozilla::MallocSizeOf mallocSizeOf)
+      : JS::RuntimeStats(mallocSizeOf)
+    {}
+
+    virtual void initExtraZoneStats(JS::Zone* zone, JS::ZoneStats* zStats)
+        override
+    {}
+
+    virtual void initExtraCompartmentStats(
+        JSCompartment* c, JS::CompartmentStats* cStats) override
+    {}
+};
+
+void Isolate::GetHeapStatistics(HeapStatistics *heap_statistics) {
+  SimpleJSRuntimeStats rtStats(moz_malloc_size_of);
+  if (!JS::CollectRuntimeStats(Runtime(), &rtStats, nullptr, false)) {
+    return;
+  }
+  heap_statistics->total_heap_size_ = rtStats.gcHeapChunkTotal;
+  heap_statistics->used_heap_size_ = rtStats.gcHeapGCThings;
```

Comment:
From what I read in the [header docs](https://dxr.mozilla.org/mozilla-central/source/js/public/MemoryMetrics.h#823), looks like what you want here is `rtStats.zTotals.sizeOfLiveGCThings() + rtStats.cTotals.sizeOfLiveGCThings()`.  But we can change this later...